dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1404,12 +1404,20 @@ public String getWorkflowDag(@ApiParam(hidden = true) @Auth Optional<User> user,
         WorkflowVersion workflowVersion = getWorkflowVersion(workflow, workflowVersionId);
         SourceFile mainDescriptor = getMainDescriptorFile(workflowVersion);
 
+        if (workflowVersion.getDagJson() != null) {
```

Comment:
Tags are not immutable, alas. If a user "recreates" a tag, this value will be of sync. It may in practice rarely or even never occur for Dockstore workflows, but it's why we created snapshots.

Options:

1. Only do this for snapshots. Pretty lame, because I assume we have very few snapshots, so most users wouldn't see a performance improvement.
2. Also track the commit sha when you store information. Compare the current commit sha with the tracked sha. Make sure to handle hosted workflows if you do this, because they don't have commit SHAs. If you go with this approach, then it shouldn't be an issue (other than space) to also cache branches. Not sure where you should track the commit id; perhaps in the dagJson column itself.
3. Whenever a version gets updated, this and the other field get set to null. This seems fragile, i.e., we could forget to clear it somewhere.